aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -52,13 +57,92 @@ static void index_select_add(const Tensor &select_indices,
   auto output_stride0 = output.stride(0);
   auto output_stride1 = output.stride(1);
 
-  for (int64_t i = 0; i < numel; i++) {
-    THBlas_axpy<T>(ddim, 1,
-            src_data + src_stride0 * select_indices_data[i], src_stride1,
-            output_data + output_stride0 * add_indices_data[i], output_stride1);
+  auto accessor = offsets.accessor<int64_t, 1>();
+  std::vector<int> lengths(offsets.numel());
```

Comment:
wait, you don't use lengths here at all, do you?

Reply:
This is a WIP. I'm trying to get the FBCode build disaster fixed before i start fixing the kernel implementation